src/components/SwitchCase/SwitchCase.tsx

```diff
@@ -1,12 +1,13 @@
 import { ReactElement } from 'react';
 
-type StringifiedValue<T> = T extends boolean
-  ? 'true' | 'false'
-  : T extends number
-    ? `${T}`
-    : T extends string
-      ? T
-      : never;
+type IsBoolean<T> = T extends boolean ? 'true' | 'false' : never;
+type IsNumber<T> = T extends number ? `${T}` : never;
+type IsString<T> = T extends string ? T : never;
+
+type StringifiedValue<T> =
+  | IsBoolean<T>
+  | IsNumber<T>
+  | IsString<T>;
```

Comment:
It's great!

But how about this approach?

```suggestion
type StringifiedValue<T> =
  | (T extends boolean ? 'true' | 'false' : never)
  | (T extends number ? `${T}` : never)
  | (T extends string ? T : never);
```

Reply:
Thx :)

This approach makes type operations clearer at a glance and more concise!